### DISCUSSION TITLE
Couple of questions regard osquery

### DISCUSSION THREAD

**Discussion by facyber** [Security-Onion-Solutions/securityonion#3853] [Unsupported Versions] [answered]
After managing to deploy osquery agents on both Windows and Linux hosts, I found some interesting results and weird things (not sure if those are issues or default behaviour). Some osquery agents were running at the same time as Wazuh agent, but now I disabled wawzuh on most of them. Here are some questions I have:

1)  By default when you install osquery, it should also ship automatically windows events and sysmon (if it is installed of course). In one of the threads I saw you wrote that sometimes it takes 15-30 minutes maybe for osquery data to show in Kibana. Is that still the case?
2) For one host after some time I start seeing less osquery data and unable to run queries on it. This is the log I found in Windows event viewer:  `Expiring events for subscriber: windows_events (overflowed limit 50000)`. Are there some limitations for osquery in Security Onion?
3) While I can confirm osquery data going well in Kibana, I can not say the same for Sysmon or Winlog Events. Sometimes I don't see them for few hours. Is that expected? Can I check something? CPU, memory, etc. are all good on manager node, so-status shows all fine.
4) If I move from Wazuh to Osquery, I see that you recommend separate node as a Fleet Manager. As I have heavy distributed architecture now (Manager + Heavy node), how would Fleet node fit in this architecture? Would osquery agent go like **Agent -> Fleet Node -> Manager Node**, or **Agent -> Fleet node -> Heavy Node -> Manager Node**? Would you recommend starting from stratch, or I can just deploy Fleet node and shutdown so-fleet on Manager?

If there are anything else I can provide for additional information, just tell me.

Cheers and thanks!

**Comment by defensivedepth** [comment]
Are you still looking for answers to these questions?

**Reply by facyber** [reply]
We moved eventualy to Wazuh as it is working without any extensive work, but I was really hoping for osquery to work mainly because of asset overview and management. So I guess I am looking answers now only for questions 2 and 4.

**Comment by defensivedepth** [comment, marked as answer]
I'll just answer all of them for posterity :)

1) Shipping WEL/sysmon is not enabled by default - you have to setup that config, docs here: https://docs.securityonion.net/en/2.3/osquery.html#shipping-windows-eventlogs

2) I'm assuming this is because the WEL were not getting expired because they were not actually being shipped off the system via a scheduled query.

3) Really would need to dig into this further when you are seeing the issue - first thing to do would be to check the Application logs on the Windows endpoint to see if there are any Launcher/osquery errors.

4) You can just deploy the Fleet Standalone. Osquery agents will connect to it, and then Filebeat will ship the osquery data over to the Manager --> Heavy
